File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.java
import java.util.stream.IntStream;
class Solution {
    public String solution(String myString, String pat) {
        return IntStream.range( 0, myString.length( ) )
						.mapToObj( idx -> myString.substring( 0, idx + 1 ) )
						.filter( sub -> sub.endsWith( pat ) )
						.reduce( "", ( acc, curr ) -> curr );
    }
}

File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.js
function solution(myString, pat) {
  // myString 의 부분문자열중 pat 으로 끝나는 문자열 리턴
  return myString
    .split("")
    .map((_, idx) => myString.slice(0, idx + 1))
    .filter((item) => item.endsWith(pat))
    .reduce((acc, curr) => curr, "");
}
